### PR TITLE
Write all samples to disk

### PR DESCRIPTION
**Pull request by igaw**
Write all samples into a file. This allows more sophisticated statistical analysis.

### CODE REVIEW

File: jitterdebugger.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	__attribute__((format(printf, 1, 2)));
+
+#define err_handler(error, fmt, arg...) do {					\
+	_err_handler(error, "%s:%s(): " fmt, __FILE__, __FUNCTION__, ## arg); \
+} while (0)
+#define warn_handler(fmt, arg...) do {						\
+	_warn_handler("%s:%s(): " fmt, __FILE__, __FUNCTION__, ## arg); \
+} while (0)
+
+
```
Comment (rralf) [inline jitterdebugger.h]: double newline

File: jitterdebugger.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#endif /* __JITTTERDEBUGGE_H */
```
Comment (rralf) [inline jitterdebugger.h]: typo

File: jitterdebugger.h [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+*/
+
+#ifndef __JITTERDEBUGGER_H
+#define __JITTERDEBUGGER_H
```
Comment (rralf) [inline jitterdebugger.h]: We don't really need include guards...

File: jitterutils.c [hunk 4]
```diff
@@ -0,0 +1,57 @@
+/*
+ * jitterdebugger - real time response messaurement tool
```
Comment (rralf) [inline jitterutils.c]: typo: measurement

File: jitterutils.c [hunk 5]
```diff
@@ -31,6 +31,87 @@
 
 #include "jitterdebugger.h"
 
+
```
Comment (rralf) [inline jitterutils.c]: double newline

File: jitterutils.c [hunk 6]
```diff
@@ -31,6 +31,87 @@
...
+static int ringbuffer_empty(uint32_t read, uint32_t write)
+{
+	return write == read;
+}
+
+static uint32_t ringbuffer_mask(uint32_t size, uint32_t idx)
+{
+	return idx & (size - 1);
+}
```
Comment (rralf) [inline jitterutils.c]: maybe inline those three guys?

File: jitterutils.c [hunk 7]
```diff
@@ -31,6 +31,87 @@
...
+
+struct ringbuffer *ringbuffer_create(unsigned int size)
+{
+	struct ringbuffer *rb;
+
+	/* Size needs to be of power of 2 */
+	if ((size & (size - 1)) != 0)
+		return NULL;
+
+	rb = calloc(1, sizeof(struct ringbuffer));
```
Comment (rralf) [inline jitterutils.c]: sizeof *rb?

File: jittersamples.c [hunk 8]
```diff
@@ -0,0 +1,123 @@
+/*
+ * jitterdebugger - real time response messaurement tool
```
Comment (rralf) [inline jittersamples.c]: typo :)

File: jittersamples.c [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+*/
```
Comment (rralf) [inline jittersamples.c]: no newline?

File: jittersamples.c [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+		printf("Missing filename\n");
+		usage();
+		exit(1);
+	}
+
+	dump_samples(argv[optind], cpuid);
+
+	return 0;
+}
+
```
Comment (rralf) [inline jittersamples.c]: trailing newline

File: jitterdebugger.h (hunk 3 above)
Comment (igaw) [reply jitterdebugger.h]: Sure, though then it doesn't hurt to have them around. If you really dislike them I can drop them for the time being, but be prepare to be screamed at when we need them :)

File: jitterutils.c (hunk 6 above)
Comment (igaw) [reply jitterutils.c]: I rather leave them as documentation there. Sure it's 'easy' to understand the expression but I still prefer the readable string.

File: jitterutils.c (hunk 7 above)
Comment (igaw) [reply jitterutils.c]: Good idea.

File: jitterdebugger.h (hunk 3 above)
Comment (rralf) [reply jitterdebugger.h]: Ok, fair enough, leave them. In another project, I was taught to leave them out whenever possible (to have an better overview of the include chain), but I'm fine with them. :-)

Comment (igaw) [reply jitterdebugger.h]: I got burned very hard by not having proper  include guards and not being self contained. I am sure there are smart people who are able to figure out all details but I am not one of those and I rather play save :)

**Pull request merged**